store/cachekv/store_bench_test.go

```diff
@@ -11,37 +9,87 @@ import (
 	"github.com/cosmos/cosmos-sdk/store/dbadapter"
 )
 
-func benchmarkCacheKVStoreIterator(numKVs int, b *testing.B) {
+// This benchmark measures the time of iterator.Next() when the parent store is blank
+func benchmarkBlankParentIteratorNext(b *testing.B, keysize int) {
+	b.StopTimer()
+
+	mem := dbadapter.Store{DB: dbm.NewMemDB()}
+	kvstore := cachekv.NewStore(mem)
+	// Use a singleton for value, to not waste time computing it
+	value := randSlice(32)
+	// Use simple values for keys, pick a random start,
+	// and take next b.N keys sequentially after.]
+	startKey := randSlice(32)
+
+	// Add 1 to avoid issues when b.N = 1
+	keys := generateSequentialKeys(startKey, b.N+1)
+	for _, k := range keys {
+		kvstore.Set(k, value)
+	}
+
 	b.ReportAllocs()
+	b.StartTimer()
+
+	iter := kvstore.Iterator(keys[0], keys[b.N])
+
+	for _ = iter.Key(); iter.Valid(); iter.Next() {
+		// TODO: Check if we need to ensure this isn't getting deadcode eliminated
```

Comment:
Let's definitely use a global sink say
```go
var sink interface{}


func benchmark*
      ....
      for _ = iter.Key(); iter.Valid(); iter.Next() {
          sink = iter
      }
```

Reply:
Interesting, didn't realize this suffices for deadcode elimination

Reply:
It does and that's the way we do so in the standard library :-)